src/Service/CaptureServiceImpl.cpp

```diff
@@ -315,6 +347,26 @@ grpc::Status CaptureServiceImpl::Capture(
     }
   }
 
+  // We need to filter out the functions instrumented by user space instrumentation.
+  CaptureOptions capture_options_without_already_instrumented_functions;
+  capture_options_without_already_instrumented_functions.CopyFrom(capture_options);
```

Comment:
Is it necessary to copy the entire CaptureOptions object here? It seems it's done only to separate the functions by instrumentation method (uprobe vs userspace). Maybe we could do the separation inside the capture option itself by adding two extra fields to hold the the function ids by instrumentation method?